src/actinia_core/resources/process_chain_monitoring.py

```diff
@@ -0,0 +1,491 @@
+# -*- coding: utf-8 -*-
+#######
+# actinia-core - an open source REST API for scalable, distributed, high
+# performance processing of geographical data that uses GRASS GIS for
+# computational tasks. For details, see https://actinia.mundialis.de/
+#
+# Copyright (c) 2016-2018 Sören Gebbert and mundialis GmbH & Co. KG
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+#######
+
+"""
+Process Chain Monitoring
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+import os
+import pickle
+import tempfile
+from flask import jsonify, make_response, Response
+from flask_restful_swagger_2 import swagger, Schema
+
+from .resource_management import ResourceManager
+from .common.response_models import ProcessingResponseModel, SimpleResponseModel
+
+__license__ = "GPLv3"
+__author__ = "Anika Weinmann"
+__copyright__ = "Copyright 2021, mundialis GmbH & Co. KG"
+__maintainer__ = "mudnialis"
+__email__ = "info@mudnialis.de"
+
+
+def create_scatter_plot(x, y, xlabel, ylabel, title):
+
+    result_file = tempfile.mktemp(suffix=".png")
```

Comment:
As this opens a code scanning issue, please follow the recommendations there by using e.g. `TemporaryFile` if applicable.